containers/Dockerfile.buildenv

```diff
@@ -354,21 +354,50 @@ RUN apt-get -y clean
 WORKDIR /work
 
 
-# We don't inherit from ci because that has lots of unnecessary compilers that
-# we don't need and would only increase the size of the image. We inherit from
-# the remote image rather than the local dev because it is faster on release
-# CI to just pull the remote dev image and build demo rather than having to
-# build all of dev and demo combined.
-# Instead of basing demo off the remote dev image, we could pre-fetch the dev
-# image and then build from the local cache like
-# FROM dev AS demo
-# Either way works, we just chose to build from remote instead.
-FROM sxscollaboration/spectre:dev AS demo
+# Deploy compiled executables to an image that can be run on HPC systems.
+# - We could also compile in the dev container and then copy the executables to
+#   a minimal deploy container to reduce its size. That requires keeping track
+#   of the shared libraries that are needed to run the executables, so to make
+#   things easier we just base the deploy container on the dev container.
+# - We inherit from the remote image rather than the local dev because it is
+#   faster on release CI to just pull the remote dev image rather than having to
+#   build the dev container again. We could pre-fetch the dev image and then
+#   build from the local cache, but that didn't work right away.
+FROM sxscollaboration/spectre:dev AS deploy
 ARG BUILDARCH
 ARG TARGETARCH
+ARG PARALLEL_MAKE_ARG=-j2
 
-# When building this image individually, the PARALLEL_MAKE_ARG from above is not
-# remembered (and it doesn't hurt to redefine the env variable).
+COPY . spectre/
+
+RUN mkdir spectre/build && cd spectre/build \
+    && cmake \
+    -D CMAKE_C_COMPILER=clang \
+    -D CMAKE_CXX_COMPILER=clang++ \
+    -D CMAKE_Fortran_COMPILER=gfortran \
+    -D CMAKE_BUILD_TYPE=Release \
+    -D DEBUG_SYMBOLS=OFF \
+    -D BUILD_PYTHON_BINDINGS=ON \
+    -D MEMORY_ALLOCATOR=SYSTEM \
+    ..
+
+# Skip compiling the executables if we are emulating the target architecture
+# because that takes a long time. They can be compiled on-demand.
+RUN if [ "$TARGETARCH" = "$BUILDARCH" ] ; then \
```

Comment:
Should we just only offer deploy on x86? I don't know of any arm clusters we'll be running on.

Reply:
The demo container is based on this, and that's built for ARM.